Add basic documentation on Kubernetes Service Binding

geoand opened the pull request:
Also changes the default value of `quarkus.kubernetes-service-binding.enabled` to `true`

Resolves: #15626

geoand committed:
Add basic documentation on Kubernetes Service Binding

Resolves: #15626

gsmet commented on docs/src/main/asciidoc/deploying-to-kubernetes.adoc:
```diff
@@ -1185,3 +1185,69 @@ The provided replicas <1>, labels <2> and environment variables <3>  were retain
+
+This list of extensions will grow as more services with supported bindings become available on Kubernetes.
+
+To enable Service Binding support, in addition to one of currently supported extensions, the `quarkus-kubernetes-service-binding` extension needs to be added to the application dependencies as well as
```
```suggestion
To enable Service Binding support, in addition to one of the currently supported extensions, the `quarkus-kubernetes-service-binding` extension needs to be added to the application dependencies as well as
```

gsmet reviewed (dismissed):
Suggested a small improvement, otherwise it looks very good!

FroMage commented on docs/src/main/asciidoc/deploying-to-kubernetes.adoc:
```diff
@@ -1185,3 +1185,69 @@ The provided replicas <1>, labels <2> and environment variables <3>  were retain
+This list of extensions will grow as more services with supported bindings become available on Kubernetes.
+
+To enable Service Binding support, in addition to one of currently supported extensions, the `quarkus-kubernetes-service-binding` extension needs to be added to the application dependencies as well as
+having the `quarkus.kubernetes-service-binding.enabled` configuration property set to `true`.
```
While this isn't about documentation, I would not have suspected this without docs.

Is there a reason why this isn't on by default when the extension is used?

geoand replied:
To be honest, I didn't remember this either and on second look it did surprise me too.
Should I change the default to true?

FroMage replied:
Yeah, after all other config sources have higher priority, so… yeah.

geoand replied:
Fixed

FroMage commented on docs/src/main/asciidoc/deploying-to-kubernetes.adoc:
```diff
@@ -1185,3 +1185,69 @@ The provided replicas <1>, labels <2> and environment variables <3>  were retain
+
+=== How does it work?
+
+When the application is deployed to an appropriately configured Kubernetes cluster, Kubernetes will create on the filesystem a set of files containing the runtime configuration (for example a file containing URL of database, another for the username).
```
```suggestion
When the application is deployed to an appropriately configured Kubernetes cluster, Kubernetes will create on the filesystem a set of files containing the runtime configuration (for example a file containing the URL of the database, another for the username).
```

geoand committed:
Change default value of quarkus.kubernetes-service-binding.enabled to true

The basic idea is that when the extension is present, an attempt to
find the service bindings makes most sense as the default behavior.
